File: plugins/autoresearch/hooks/sky-auto-auth.js
#!/usr/bin/env node
/**
 * PreToolUse Hook: SkyPilot Auto-Authorization
 *
 * Auto-authorizes SkyPilot commands (sky launch, sky exec, sky logs, etc.)
 * without requiring manual permission prompts. Modeled after CCG-workflow's
 * codeagent-wrapper auto-authorization hook.
 *
 * Exit codes:
 *   0 - Allow (always, for matching commands outputs permission grant)
 */

const MAX_STDIN = 1024 * 1024;

const SKY_PATTERNS = [
  'sky launch',
  'sky exec',
  'sky logs',
  'sky status',
  'sky down',
  'sky cancel',
  'sky stop',
  'sky start',
  'sky queue',
  'sky jobs',
];

let data = '';
process.stdin.setEncoding('utf8');

process.stdin.on('data', chunk => {
  if (data.length < MAX_STDIN) data += chunk.substring(0, MAX_STDIN - data.length);
});

process.stdin.on('end', () => {
  try {
    const disabled = (process.env.DISABLED_HOOKS || '').split(',');
    if (disabled.includes('sky-auto-auth')) {
      process.stdout.write(data);
      return;
    }

    const input = JSON.parse(data);
    const command = input.tool_input?.command || '';

    const isSkyCommand = SKY_PATTERNS.some(p => command.includes(p));

    if (isSkyCommand) {
      const authResponse = JSON.stringify({
        hookSpecificOutput: {
          hookEventName: 'PreToolUse',
          permissionDecision: 'allow',
          permissionDecisionReason: 'SkyPilot command auto-approved by autoresearch plugin'
        }
      });
      process.stdout.write(authResponse);
      return;
    }

    process.stdout.write(data);
  } catch (err) {
    console.error(`[Hook] sky-auto-auth error: ${err.message}`);
    process.stdout.write(data);
  }
});

File: plugins/autoresearch/hooks/sky-auto-auth.py
#!/usr/bin/env python3
"""
PreToolUse Hook: SkyPilot Auto-Authorization

Auto-authorizes SkyPilot commands (sky launch, sky exec, sky logs, etc.)
without requiring manual permission prompts.

Exit codes:
  0 - Allow (always, for matching commands outputs permission grant)
"""

import json
import os
import sys

MAX_STDIN = 1024 * 1024

SKY_PATTERNS = [
    "sky launch",
    "sky exec",
    "sky logs",
    "sky status",
    "sky down",
    "sky cancel",
    "sky stop",
    "sky start",
    "sky queue",
    "sky jobs",
]


def main():
    data = sys.stdin.read(MAX_STDIN)

    try:
        disabled = os.environ.get("DISABLED_HOOKS", "").split(",")
        if "sky-auto-auth" in disabled:
            sys.stdout.write(data)
            return

        inp = json.loads(data)
        command = inp.get("tool_input", {}).get("command", "")

        is_sky_command = any(p in command for p in SKY_PATTERNS)

        if is_sky_command:
            auth_response = json.dumps({
                "hookSpecificOutput": {
                    "hookEventName": "PreToolUse",
                    "permissionDecision": "allow",
                    "permissionDecisionReason": "SkyPilot command auto-approved by autoresearch plugin",
                }
            })
            sys.stdout.write(auth_response)
            return

        sys.stdout.write(data)
    except Exception as err:
        print(f"[Hook] sky-auto-auth error: {err}", file=sys.stderr)
        sys.stdout.write(data)


if __name__ == "__main__":
    main()
